functorch/_src/aot_autograd.py

```diff
@@ -567,7 +1105,106 @@ def backward(ctx, *flat_args):
+            # TODO: think harder about what happens if (a view of) one of these mutated input views is ALSO returned
+            new_inputs, metadata = merge_view_inputs(no_dupe_args, CompiledFunction.fw_metadata.mutated_input_info)
+            # We're just re-running the original-args-to-synthetic-base transformation
+            # that we ran during compilation.
+            # This returns metadata that we use during tracing to recover the input views,
+            # which we don't actually need at runtime.
+            assert metadata is not None
+            no_dupe_args_with_synthetic_bases = new_inputs
+        else:
+            no_dupe_args_with_synthetic_bases = no_dupe_args
+
+        outs = CompiledFunction.apply(*no_dupe_args_with_synthetic_bases)
+
+        # Step 3: After running the compiled fw, apply updates to mutated inputs
+        if CompiledFunction.num_mutated_inputs > 0:
+            assert len(outs) == CompiledFunction.num_mutated_data_inputs + CompiledFunction.num_non_aliased_outs
+            # Calling convention: mutated inputs show up first.
+            updated_inputs = outs[:CompiledFunction.num_mutated_data_inputs]
+            fw_outs = outs[CompiledFunction.num_mutated_data_inputs:]
+            curr_mutated_inpt_idx = 0
+            curr_metadata_only_mutated_inpt_idx = 0
+            for inpt_idx, mutation_type in enumerate(CompiledFunction.fw_metadata.mutated_input_info):
+                if mutation_type == MutationType.none:
+                    continue
+                original_inpt = no_dupe_args[inpt_idx]
+                if mutation_type == MutationType.metadata_only:
+                    expected_metadata = CompiledFunction.fw_metadata\
+                        .metadata_mutation_input_info[curr_metadata_only_mutated_inpt_idx]
+                    curr_metadata_only_mutated_inpt_idx += 1
+                    original_inpt.as_strided_(
+                        expected_metadata.size(), expected_metadata.stride(), expected_metadata.storage_offset())
+                else:
+                    updated_inpt = updated_inputs[curr_mutated_inpt_idx]
+                    curr_mutated_inpt_idx += 1
+                    # TODO: handle resize_() on inputs to a larger size.
+                    # This is actually non-trivial to detect, so we should probably just handle it
+                    # (or make dynamo detect).
+                    # We can't just check of original_inpt.storage_size != updated_inpt.storage_size,
+                    # Because the original_inpt might be a view of some larger tensor,
+                    # and updated_inpt is always densely packed.
+                    if original_inpt.size() != updated_inpt.size() \
+                            or original_inpt.stride() != updated_inpt.stride() \
+                            or original_inpt.storage_offset() != updated_inpt.storage_offset():
```

Comment:
I'm surprised this check can't be precomputed

Reply:
oh this is probably thing I mentioned about separating metadata and data mutation flags

Reply:
yep - fixed in upcoming PR